File: Lista03_HerancaEPolimorfismo/PegadaCarbono/app/src/main/java/PegadaCarbono/Casa.java
package PegadaCarbono;

public class Casa extends Predio{
    public int nComodos; 
    public int nChuveiros; 

    public Casa(int nFrequentadores, double mediaLixoPorFrequentador, double consumoGasPorDia, int nComodos, int nChuveiros) {
        super(nFrequentadores, mediaLixoPorFrequentador, consumoGasPorDia);
        this.nComodos = nComodos;
        this.nChuveiros = nChuveiros;
    }

    public int getNComodos() {
        return nComodos;
    }

    public void setNComodos(int nComodos) {
        this.nComodos = nComodos;
    }

    public int getNChuveiros() {
        return nChuveiros;
    }

    public void setNChuveiros(int nChuveiros) {
        this.nChuveiros = nChuveiros;
    }
}

File: Lista11_Python/Ex03_PegadaDeCarbono/casa.py
from predio import *

class Casa(Predio):

    def __init__(self, nFrequentadores, mediaLixoPorFrequentador, consumoGasDiario, nComodos, nChuveiros):
        super().__init__(nFrequentadores, mediaLixoPorFrequentador, consumoGasDiario)
        self.nComodos = nComodos
        self.nChuveiros = nChuveiros

    def __str__(self):
        strComodos = str(self.nComodos)
        strChuveiros = str(self.nChuveiros)

        string = super().__str__()
        string = string+"\nnComodos: "+strComodos+"\n"
        string = string+"nComodos: "+strChuveiros

        return string
    
    def getNComodos(self):
        return self.nComodos

    def getPegadaCarbono(self):
        mediaLixoPorFrequentador = super().getMediaLixoPorFrequentador()
        nFrequentadores = super().getNFrequentadores()
        consumoGasPorDia = super().getConsumoGasDiario()
        pCarbono = (mediaLixoPorFrequentador * nFrequentadores * 0.1) + (consumoGasPorDia * 0.1)
        return pCarbono
